### DISCUSSION TITLE
Emit events when playing through any method (including Bluetooth)

### DISCUSSION THREAD

**Discussion by Joshfindit** [sle118/squeezelite-esp32#195] [Ideas]
I'm looking for the end goal of being able to automate lighting/smart outlets based on whether the squeezelite is playing audio (device connected but not playing is fine too).

I do have access to the SLIM event subscription so it seems like a handy method. I can also make any type of network API work (push is desired over poll, but whatever)

**Comment by sle118** [comment]
Not sure about how this could be implemented. We're pushing the limits of the platform already, and adding yet another socket is going to cost some memory.

If you look at the network traffic between a browser and a player, you'll see that the status message contains two Bluetooth related fields.

Perhaps you could build your automation from two different sources: the Logitech media server for playback originating from there, and the player status JSON for Bluetooth.

https://github.com/sle118/squeezelite-esp32/blob/a2c16decee56126419d46f74e62be60261d8c683/components/wifi-manager/webapp/src/js/custom.js#L467

**Comment by Joshfindit** [comment]
I’ll take a look at that network request the next chance I get, but I also wonder:

Could Bluetooth events be emitted to the SLIM server? That way the memory could be saved

**Comment by sle118** [comment]
I doubt that the slimproto has any provision for that sort of event, unfortunately. If you think of something specific, let me know

**Comment by sle118** [comment]
I'm converting this to a discussion for now

**Comment by pmannk** [comment]
This is definitely not a fix for your request, it's merely a workaround.
I was looking for something similar and ended up building an automation in HA which leverages a couple of conditions:

- The "source" attribute from the spotify integration (ie, when spotify changes to playing and the source is my esp32 it triggers) 
- The "connected_paired_devices" attribute from the mobile app integration (bluetooth connection sensor). When the address of the esp32 appears in the connected devices it triggers.

This is really messy since I had to build triggers for 3 different phones and 2 different spotify integrations, but it's the best I could do as a workaround.
